pymeasure/instruments/siglenttechnologies/siglent_spdbase.py

```diff
@@ -221,30 +217,15 @@ def __init__(self, instrument,
         dynamic=True
     )
 
-    def ask(self, cmd):
-        return self.instrument.ask(cmd.format(channel=self.channel))
-
-    def write(self, cmd):
-        self.instrument.write(cmd.format(channel=self.channel))
-
-    def values(self, cmd, **kwargs):
-        """ Reads a set of values from the instrument through the adapter,
-        passing on any key-word arguments.
-        """
-        return self.instrument.values(cmd.format(channel=self.channel), **kwargs)
-
-    def check_errors(self):
-        return self.instrument.check_errors()
-
     def enable_output(self, enable: bool = True):
         """Enable the channel output.
 
         :type: bool
             ``True``: enables the output
             ``False``: disables it
         """
-        self.instrument.selected_channel = self.channel
-        self.write(f'OUTP CH{self.channel},{("OFF","ON")[enable]}')
+        self.parent.selected_channel = self.id
+        self.write('OUTP CH{ch},' + ("OFF", "ON")[enable])
```

Comment:
Interesting usage.
Afaik, True is not guaranteed to be equal to 1, therefore I suggest:
```suggestion
        self.write('OUTP CH{ch},' + ("ON" if enable else "OFF"))
```

Reply:
I've been using this construction myself for a while in Python now. I've found it to be quite a neat way of differentiating between 2 states when constructing strings. I understand the confusion though, I spent the other half of my days working in C/C++ where true can often be anything other than 0

Reply:
I know it works, but I find it clearer if spelled out (`"ON" if enable else "OFF"`).
The other way reminds me to of using implicit casts from bool to int/float in C++ like `somevalue = x + somefunc(y)*(foo >= bar)` - I don't find that very readable, either.

Reply:
I've changed this over now, and also in the other places Id used the same construction